### DISCUSSION TITLE
[Feature] Button - Delete on Mobile Device only

### DISCUSSION THREAD

**Discussion by zhdenny** [immich-app/immich#2190] [Feature Request]
### The feature

In the app on Mobile device, the Delete button specifically Deletes an item from BOTH Immich Server and Device.

Can we have a Button to Delete an item on the device ONLY?

This would allow us to easily free up space on our Mobile devices all from within the Mobile Immich App. Currently, I have to perform this action outside the Immich App.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by dogs-1985** [comment]
> ### The feature
> In the app on Mobile device, the Delete button specifically Deletes an item from BOTH Immich Server and Device.
> 
> Can we have a Button to Delete an item on the device ONLY?
> 
> This would allow us to easily free up space on our Mobile devices all from within the Mobile Immich App. Currently, I have to perform this action outside the Immich App.
> 
> ### Platform
> * [ ]  Server
> * [ ]  Web
> * [x]  Mobile

Hi, 
I wish to have these features. But, if you manually delete the files. Do you face some problems like the blank images on the app screen? I need to log out and log in to the app again only can solve this problem.


Hi developer, 
I also wish to have an additional feature too.
After uploading the files to the Immich server, can I have a button to free my storage space? This means deleting my local file. 
Or, can I have a scheduler that frees my storage space when the photos are uploaded and more than certain days like 30 days?

**Comment by auanasgheps** [comment]
I would like to have this feature too, following the logic to "Free Up space" on devices

**Comment by wackyesolution** [comment]
I'd really like to be able to free up space on my device automatically after the upload is complete. as say @auanasgheps 

I think it's an essential function because it allows you to clean the gallery with the automatic certainty that the media has been successfully uploaded.

I hope it will be considered and implemented as soon as possible.

Can any developer tell me whether this feature is planned or not? If the answer is yes, I would like to know if it will also work with already uploaded media or is it better not to upload anything so as to wait for the possible implementation of this essential function to be added?

Thank you for all the amazing work done 💚

**Reply by jrasm91** [reply]
It is planned, but not started yet. I'm pretty sure the implementation would work regardless when the photos were uploaded as this would most likely be a client-side only change.

**Comment by bo3OU** [comment]
Agree, as the phone will upload pictures again if you delete from server alone.
Might I also suggest to not upload pictures again if it was uploaded to the server and deleted from server with web

**current :** 
- take photo
- photo is uploaded
- delete photo from server
- open app on mobile
- photo is uploaded again

**wanted :** 
- take photo
- photo is uploaded
- delete photo from server (though web)
- open app on mobile
- photo does not get uploaded

**Comment by bo0tzz** [comment]
I believe this is implemented now.
